### PR TITLE
fix: Copybooks mismatch in cache #1328

### PR DESCRIPTION
**Pull request by ishche**
I've added a documentUri of a program to copybook cache id to distinguish copybooks based on a program it is used in.

### CODE REVIEW

File: server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/CopybookHierarchy.java [hunk 1]
```diff
@@ -73,6 +73,12 @@ public void pop() {
     recursiveReplaceStmtStack.pollFirst();
   }
 
+  /** @return root document URI */
+  public String getRootDocumentUri() {
```
Comment (grianbrcom) [inline server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/CopybookHierarchy.java]: How about making it Optional<String>?

File: server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/copybooks/analysis/CopybookName.java [hunk 2]
```diff
@@ -38,11 +38,11 @@ public CopybookName(String displayName, String dialectType) {
...
   /**
-   * Creates a processing copybook name using a name and a dialect type
-   *
+   * Creates a processing copybook name (cache id) using a name and a dialect type
+   * @param documentUri Document with COPY instruction for this copybook.
    * @return a processing copybook name
    */
-  public String getProcessingName() {
-    return String.format("%s#%s", qualifiedName, dialectType);
+  public String getProcessingName(String documentUri) {
```
Comment (grianbrcom) [inline server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/copybooks/analysis/CopybookName.java]: Why `documentUri` is not a part of this `CopybookName`?

Comment (ishche) [reply server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/copybooks/analysis/CopybookName.java]: CopybookName can be called from different documents, so it's a parameter for a cache key. (moved to cache service now)

File: server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/CopybookHierarchy.java (hunk 1 above)
Comment (ishche) [reply server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/CopybookHierarchy.java]: Make sense

**Review by grianbrcom — APPROVED** [review]
🏒

**Pull request merged**